### PR TITLE
Ensure alpine and arm SKUs have powershell.config.json file with experimental features enabled

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

We needed to invoke pwsh to generate the list of experimental features. Hence, it did not work on cross compiled packages like ARM and Alpine. Now we have static file with the experimental feature list. This is used as the source of truth for listing the experimental feature to be included in the powershell.config.json.

Updated build configurations to use the experimental files json.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: tools/UpdateDotnetRuntime.ps1 [hunk 1]
```diff
@@ -342,7 +342,7 @@ if ($dotnetUpdate.ShouldUpdate) {
         Import-Module "$PSScriptRoot/../build.psm1" -Force
         Import-Module "$PSScriptRoot/packaging" -Force
         Start-PSBootstrap -Package
-        Start-PSBuild -Clean -Configuration Release -CrossGen -InteractiveAuth:$InteractiveAuth
+        Start-PSBuild -Clean -Configuration Release -CrossGen -InteractiveAuth:$InteractiveAuth -ExperimentalFeatureJsonFilePath "$PSScriptRoot/../ExperimentalFeatures.json"
```
Comment (PaulHigin) [inline tools/UpdateDotnetRuntime.ps1:345]: If the `ExperimentalFeatures.json` file is always at the same build path location, do we need to pass in the file path?

**Review by PaulHigin — APPROVED** [review]

File: ExperimentalFeatures.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "PSCleanBlock",
+  "PSCommandNotFoundSuggestion",
+  "PSLoadAssemblyFromNativeCode",
+  "PSNativeCommandArgumentPassing",
+  "PSNativeCommandErrorActionPreference",
+  "PSNativePSPathResolution",
+  "PSRemotingSSHTransportErrorHandling",
+  "PSSubsystemPluginModel",
+  "PSAMSIMethodInvocationLogging"
+]
```
Comment (daxian-dbw) [inline ExperimentalFeatures.json:12]: If there is no automation to keep this in sync with code, it's probably will be out of sync sooner or later.

File: build.psm1 [hunk 3]
```diff
@@ -624,36 +625,50 @@ Fix steps:
...
+            throw "ExperimentalFeatureJsonFilePath: $ExperimentalFeatureJsonFilePath does not exist"
         }
 
+        $json = Get-Content -Raw $ExperimentalFeatureJsonFilePath
         $config += @{ ExperimentalFeatures = ([string[]] ($json | ConvertFrom-Json)) }
     }
+    else {
+        # When building preview, we want the configuration to enable all experiemental features by default
+        # ARM is cross compiled, so we can't run pwsh to enumerate Experimental Features
+        if (-not $SkipExperimentalFeatureGeneration -and
```
Comment (daxian-dbw) [inline build.psm1:643]: If that JSON file is always available, do we need this `else` code path?

File: tools/UpdateDotnetRuntime.ps1 (hunk 1 above)
Comment (adityapatwardhan) [reply tools/UpdateDotnetRuntime.ps1]: Fixed.. always use the files that are checked in

File: ExperimentalFeatures.json (hunk 2 above)
Comment (adityapatwardhan) [reply ExperimentalFeatures.json]: Added automation via GitHub Actions in PR #16838

File: build.psm1 (hunk 3 above)
Comment (adityapatwardhan) [reply build.psm1]: Removed

**Comment by adityapatwardhan** [comment]
@TravisEz13 @daxian-dbw Please re-review

File: build.psm1 [hunk 4]
```diff
@@ -619,40 +619,29 @@ Fix steps:
...
-            $ExperimentalFeaturesFromGalleryModulesInPSHome = @()
-            $ExperimentalFeaturesFromGalleryModulesInPSHome | ForEach-Object {
-                if (!$expFeatures.Contains($_)) {
-                    $null = $expFeatures.Add($_)
-                }
+        -not (Test-IsReleaseCandidate $psVersion)
+    ) {
+
+        if ($Options.Runtime -like "*win*") {
+            $config = @{ "Microsoft.PowerShell:ExecutionPolicy" = "RemoteSigned";
```
Comment (TravisEz13) [inline build.psm1:629]:
```suggestion
			# Execution Policy is only supported on Windows
            $config = @{ "Microsoft.PowerShell:ExecutionPolicy" = "RemoteSigned";
```

**Review by TravisEz13 — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**